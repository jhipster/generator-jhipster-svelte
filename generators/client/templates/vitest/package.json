{
	"scripts": {
		"test": "vitest run --coverage",
 		"test:ui": "vitest --ui --coverage",
 		"test:watch": "vitest"
	},
	"devDependencies": {
		"@vitest/ui": "0.34.6",
 		"@vitest/coverage-v8": "1.0.0-beta.0",
		"jsdom": "22.1.0",
		"vitest" : "0.34.6",
		"vitest-sonar-reporter": "0.5.0"
	}
}

