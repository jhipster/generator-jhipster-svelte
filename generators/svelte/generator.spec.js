import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'svelte';
const SUB_GENERATOR_NAMESPACE = `jhipster-svelte:${SUB_GENERATOR}`;

const entitySimple = {
	name: 'Simple',
	changelogDate: '20220129000100',
	jpaMetamodelFiltering: true,
	fields: [{ fieldName: 'simpleName', fieldType: 'String' }],
};

const entityAnotherSimple = {
	name: 'AnotherSimple',
	changelogDate: '20220129000200',
	fields: [{ fieldName: 'simpleName', fieldType: 'String' }],
	dto: 'mapstruct',
	service: 'serviceImpl',
	pagination: 'pagination',
	clientRootFolder: 'test-root',
};

const entitySkipClient = {
	name: 'SkipClient',
	changelogDate: '20220129000400',
	skipClient: true,
	fields: [{ fieldName: 'skipClientName', fieldType: 'String' }],
};

const entityWithEnum = {
	name: 'SkipClient',
	changelogDate: '20220129000400',
	skipClient: true,
	fields: [{ fieldName: 'skipClientName', fieldType: 'MyEnum', fieldValues: 'FRENCH,ENGLISH' }],
};

describe('SubGenerator svelte of svelte JHipster blueprint', () => {
	describe('run', () => {
		beforeAll(async function () {
			await helpers
				.run(SUB_GENERATOR_NAMESPACE)
				.withJHipsterConfig({}, [entitySimple, entityAnotherSimple, entitySkipClient, entityWithEnum])
				.withOptions({
					ignoreNeedlesError: true,
					blueprints: 'svelte',
				})
				.withJHipsterLookup()
				.withParentBlueprintLookup();
		});

		it('should succeed', () => {
			expect(result.getStateSnapshot()).toMatchSnapshot();
		});
	});
});
