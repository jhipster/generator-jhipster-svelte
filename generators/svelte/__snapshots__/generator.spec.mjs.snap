// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator svelt of svelte JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "template-file-svelt": {
    "stateCleared": "modified",
  },
}
`;
